### DISCUSSION TITLE
Delay of emitting message from frontend to backend

### DISCUSSION THREAD

**Discussion by SAIREDDY54** [miguelgrinberg/Flask-SocketIO#1918] [Q&A]
I am using flask as backend and reactjs as frontend, on button click I am emitting data from frontend to backend but request is delayed and when I inspect the traffic it shows pending. I am serving the frontend on localhost.

Please look into this.

![screenshot](https://pasteboard.co/t5kiZFR5EtTR.png)

**Comment by SAIREDDY54** [comment]
Image link

https://pasteboard.co/t5kiZFR5EtTR.png

**Comment by miguelgrinberg** [comment]
I can't really tell you much with the information that is visible in your picture, but pending requests are normal, those are part of the long polling protocol used by Socket.IO. As a side note, you should configure your server so that it also supports WebSocket, which performs better than long-polling.

The picture also shows that your browser has multiple connections to the server, which is very likely a bug in your application and not something you intended. Normally each client start a single Socket.IO connection, not many.
